src/Avalonia.Base/AvaloniaObject.cs

```diff
@@ -860,6 +868,28 @@ private void LogPropertySet<T>(AvaloniaProperty<T> property, T value, BindingPri
                 priority);
         }
 
+        internal string GetDebugDisplay(bool includeContent)
+        {
+            var builder = new StringBuilder();
+            BuildDebugDisplay(builder, includeContent);
+            return builder.ToString();
+        }
+
+        internal virtual void BuildDebugDisplay(StringBuilder builder, bool includeContent)
+        {
+            var type = GetType();
+
+            if (type.Namespace is { } ns &&
+                (ns == "Avalonia" || ns.StartsWith("Avalonia.", StringComparison.Ordinal)))
+            {
+                builder.Append(type.Name);
+            }
+            else
+            {
+                builder.Append(ToString());
```

Comment:
`ToString()` might return unwanted output, if it was overridden.
I think expected behavior would be to return type.FullName here.

Reply:
It was intended: by default if a type doesn't have a `DebuggerDisplay`, the debugger defaults to `ToString`.

But since we now have this attribute on the base `AvaloniaObject`, custom types outside of the Avalonia solution with `ToString` overridden don't get that string displayed in the debugger anymore: this code path intends to fix that.